update-ubuntu-or-debian-server/run.sh

```
@@ -2,193 +2,928 @@
+            sleep 5
+            ((attempt++))
+        fi
+    done
+    
+    FAILED_OPERATIONS+=("$operation")
+    return 1
+}
+
+# Function to check if in maintenance window
+in_maintenance_window() {
+    local current_time=$(date +%H:%M)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)